src/wp-includes/functions.php

```diff
@@ -3399,6 +3399,7 @@ function wp_nonce_ays( $action ) {
  *                                  is a WP_Error.
  *     @type bool   $exit           Whether to exit the process after completion. Default true.
  * }
+ * @phpstan-return never
```

Comment:
I wanted to add `@return never` but didn't for the reason you pointed out. PHPStan supports it though, regardless of PHP version. It's particularly useful here because static analysis cannot determine that this function always exits.

Reply:
I would much rather we start the discussion on changing the WP Docs guideline about the `@return` tag not being mandatory.

I wrote up a proposal about this quite a while back, but then the RFC repo to which it was supposed to be pulled got closed, so I guess we'd have to go through a Make post to get it discussed now.

If you are interested, this was the initial draft proposal: https://github.com/jrfnl/rfcs/blob/feature/docs-always-demand-return-tag/text/0000-docs-always-use-return-tag.md
And yes, the proposal would need some minor updates for current PHP before it could be published in one way or another if we'd go that route.